Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITestCocoa.mm

```diff
@@ -438,8 +438,17 @@
 
     auto testComplete = [this, protectedThis = Ref { *this }, test](JSValue *result, JSValue *error) {
         if (error || m_hitAssertion) {
-            JSValue *errorMessageValue = error.isObject && [error hasProperty:@"message"] ? error[@"message"] : error;
-            WebProcess::singleton().send(Messages::WebExtensionController::TestFinished(test.testName, false, adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", debugString(errorMessageValue)]).get(), test.location.first, test.location.second), test.webExtensionControllerIdentifier);
+            NSString *errorMessage;
+            if (error) {
+                JSValue *errorMessageValue = error.isObject && [error hasProperty:@"message"] ? error[@"message"] : error;
+                errorMessage = adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", debugString(errorMessageValue)]).get();
+            } else if (!m_assertionMessage.isNull())
+                errorMessage = adoptNS([[NSString alloc] initWithFormat:@"Promise rejected with an error: %@", m_assertionMessage.createNSString().get()]).get();
+            else
+                errorMessage = @"Promise rejected without an error";
+
+            WebProcess::singleton().send(Messages::WebExtensionController::TestFinished(test.testName, false, errorMessage, test.location.first, test.location.second), test.webExtensionControllerIdentifier);
```

Comment:
i'm not seeing this extra newline locally, which is strange

Reply:
I think the newline is fine.